src/components/Map/CircleComponent.tsx

```diff
@@ -1,23 +1,45 @@
+/* eslint-disable multiline-ternary */
 import L from 'leaflet';
-import { Circle, Marker } from 'react-leaflet';
+import { useRef } from 'react';
+import { Circle, Marker, Popup } from 'react-leaflet';
+import { type SensorData } from 'src/models/SensorData';
+
+import PopupComponent from './PopupComponent';
 
 interface CircleComponentProps {
-  center: [number, number]
-  radius: number
-  text: string
-  color: string
+  center: [number, number];
+  radius: number;
+  text: string;
+  color: string;
+  sensorData: SensorData;
 }
 
-function CircleComponent({ center, radius, text, color }: CircleComponentProps) {
+function CircleComponent({
+  center,
+  radius,
+  text,
+  color,
+  sensorData
+}: CircleComponentProps) {
+  const circleRef = useRef<L.Circle>(null);
   const text1 = L.divIcon({ className: 'text-2xl font-bold', html: text });
 
+  const onClick = () => {
+    circleRef.current?.openPopup();
+  };
```

Comment:
This function is quite short so you can make it inline and put `() => circleRef.current?.openPopup()` directly as a value of the `click` property in the `eventHandlers` object.

It's just my preference for short 1-line functions. If you fill more comfortable with your solution, just stick to it.